x/profiles/types/models_chain_links.go

```diff
@@ -109,7 +145,114 @@ func (p Proof) Verify(unpacker codectypes.AnyUnpacker, address AddressData) erro
 // UnpackInterfaces implements codectypes.UnpackInterfacesMessage
 func (p *Proof) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
 	var pubKey cryptotypes.PubKey
-	return unpacker.UnpackAny(p.PubKey, &pubKey)
+	err := unpacker.UnpackAny(p.PubKey, &pubKey)
+	if err != nil {
+		return err
+	}
+
+	var signatureData SignatureData
+	return unpacker.UnpackAny(p.Signature, &signatureData)
+}
+
+// --------------------------------------------------------------------------------------------------------------------
+
+// SignatureData represents a generic single- or multi- signature data
+type SignatureData interface {
+	proto.Message
+
+	isSignatureData()
+}
+
+// SignatureDataToCosmosSignatureData allows to convert the given SignatureData to a Cosmos SignatureData instance
+// unpacking the proto.Any instance using the given unpacker.
+func SignatureDataToCosmosSignatureData(unpacker codectypes.AnyUnpacker, s SignatureData) (signing.SignatureData, error) {
+	switch data := s.(type) {
+	case *SingleSignatureData:
+		return &signing.SingleSignatureData{
+			Signature: data.Signature,
+			SignMode:  data.Mode,
+		}, nil
+
+	case *MultiSignatureData:
+		sigs, err := unpackSignatures(unpacker, data.Signatures)
+		if err != nil {
+			return nil, err
+		}
+
+		return &signing.MultiSignatureData{
+			BitArray:   data.BitArray,
+			Signatures: sigs,
+		}, nil
+	}
+
+	return nil, fmt.Errorf("signature type not supported: %T", s)
+}
+
+func SignatureDataFromCosmosSignatureData(data signing.SignatureData) SignatureData {
+	switch data := data.(type) {
+	case *signing.SingleSignatureData:
+		return &SingleSignatureData{
+			Mode:      data.SignMode,
+			Signature: data.Signature,
+		}
+	case *signing.MultiSignatureData:
+		sigAnys := make([]*codectypes.Any, len(data.Signatures))
+		for i, data := range data.Signatures {
+			sigAny, _ := codectypes.NewAnyWithValue(SignatureDataFromCosmosSignatureData(data))
```

Comment:
We should check the error here to make sure it's `nil` and the signature is encoded properly. The `SignatureDataFromCosmosSignatureData` should return `(SignatureData, error)`